### ISSUE TITLE
How to reproduce the same visual results as in your paper？

### ISSUE THREAD

**Issue by Stenwarte** [ShuhangGu/DASR#8]
When training the models with the given parameters, we can get the best SR visual results almost like this:
![image](https://user-images.githubusercontent.com/82566391/114824980-387b7f80-9df8-11eb-8bc5-8cb5ce4cfd2d.png)
They are not as clear as your results. And as the number of iterations increases, there will be more artifacts like this：
![image](https://user-images.githubusercontent.com/82566391/114826013-86dd4e00-9df9-11eb-9701-e4bb4d5c4389.png)
So is our reproduction normal or is there something wrong？

**Comment by greatlog** [comment]
Recently I found that all unpaired SR methods accepted by these conferences can not be reimplemented.

**Comment by 543877815** [comment]
Have you retrained the DSN to reproduce the whole experiment?
I think the first image is a good result for blind SR with unpaired images.

**Comment by mrgreen3325** [comment]
> Have you retrained the DSN to reproduce the whole experiment? I think the first image is a good result for blind SR with unpaired images.

Hello.
Have you try this?
Is it successful to reproduce the whole experiment?

**Comment by 543877815** [comment]
> > Have you retrained the DSN to reproduce the whole experiment? I think the first image is a good result for blind SR with unpaired images.
> 
> Hello. Have you try this? Is it successful to reproduce the whole experiment?

No. In my understanding, you need to train the DSN carefully in order to generate LR that is degradation consistent with real-world LR. Only in this way, the SRN trained on the generated LR and HR pairs might have good generalization on real-world LR.

**Comment by mrgreen3325** [comment]
> > > Have you retrained the DSN to reproduce the whole experiment? I think the first image is a good result for blind SR with unpaired images.
> > 
> > 
> > Hello. Have you try this? Is it successful to reproduce the whole experiment?
> 
> No. In my understanding, you need to train the DSN carefully in order to generate LR that is degradation consistent with real-world LR. Only in this way, the SRN trained on the generated LR and HR pairs might have good generalization on real-world LR.

Thanks.
Do you have any tricks to train the DSN well?
I am training on it, however, the generated LR contains some artifact on it.

**Comment by 543877815** [comment]
> > > > Have you retrained the DSN to reproduce the whole experiment? I think the first image is a good result for blind SR with unpaired images.
> > > 
> > > 
> > > Hello. Have you try this? Is it successful to reproduce the whole experiment?
> > 
> > 
> > No. In my understanding, you need to train the DSN carefully in order to generate LR that is degradation consistent with real-world LR. Only in this way, the SRN trained on the generated LR and HR pairs might have good generalization on real-world LR.
> 
> Thanks. Do you have any tricks to train the DSN well? I am training on it, however, the generated LR contains some artifact on it.

What do you mean by "artifact" on LR?

**Comment by mrgreen3325** [comment]
> > > > > Have you retrained the DSN to reproduce the whole experiment? I think the first image is a good result for blind SR with unpaired images.
> > > > 
> > > > 
> > > > Hello. Have you try this? Is it successful to reproduce the whole experiment?
> > > 
> > > 
> > > No. In my understanding, you need to train the DSN carefully in order to generate LR that is degradation consistent with real-world LR. Only in this way, the SRN trained on the generated LR and HR pairs might have good generalization on real-world LR.
> > 
> > 
> > Thanks. Do you have any tricks to train the DSN well? I am training on it, however, the generated LR contains some artifact on it.
> 
> What do you mean by "artifact" on LR?
Thanks.
Something like the line or dot on LR, and some color shift.
![01_116_3013](https://user-images.githubusercontent.com/25168884/174220717-1612cf09-82da-43f5-bc67-fee3ed454b48.png)
My HR is face dataset, while LR is some images taken by my phone camera.

**Comment by 543877815** [comment]
> > > > > > Have you retrained the DSN to reproduce the whole experiment? I think the first image is a good result for blind SR with unpaired images.
> > > > > 
> > > > > 
> > > > > Hello. Have you try this? Is it successful to reproduce the whole experiment?
> > > > 
> > > > 
> > > > No. In my understanding, you need to train the DSN carefully in order to generate LR that is degradation consistent with real-world LR. Only in this way, the SRN trained on the generated LR and HR pairs might have good generalization on real-world LR.
> > > 
> > > 
> > > Thanks. Do you have any tricks to train the DSN well? I am training on it, however, the generated LR contains some artifact on it.
> > 
> > 
> > What do you mean by "artifact" on LR?
> > Thanks.
> > Something like the line or dot on LR, and some color shift.
> > ![01_116_3013](https://user-images.githubusercontent.com/25168884/174220717-1612cf09-82da-43f5-bc67-fee3ed454b48.png)
> > My HR is face dataset, while LR is some images taken by my phone camera.

Can you show some real-world LR images that are from your phone?  If the real-world LR has the same artifact, it works. 
The color shift is normal since the training is a balance between content loss and gan loss in my understanding.

**Comment by mrgreen3325** [comment]
> > > > > > > Have you retrained the DSN to reproduce the whole experiment? I think the first image is a good result for blind SR with unpaired images.
> > > > > > 
> > > > > > 
> > > > > > Hello. Have you try this? Is it successful to reproduce the whole experiment?
> > > > > 
> > > > > 
> > > > > No. In my understanding, you need to train the DSN carefully in order to generate LR that is degradation consistent with real-world LR. Only in this way, the SRN trained on the generated LR and HR pairs might have good generalization on real-world LR.
> > > > 
> > > > 
> > > > Thanks. Do you have any tricks to train the DSN well? I am training on it, however, the generated LR contains some artifact on it.
> > > 
> > > 
> > > What do you mean by "artifact" on LR?
> > > Thanks.
> > > Something like the line or dot on LR, and some color shift.
> > > ![01_116_3013](https://user-images.githubusercontent.com/25168884/174220717-1612cf09-82da-43f5-bc67-fee3ed454b48.png)
> > > My HR is face dataset, while LR is some images taken by my phone camera.
> 
> Can you show some real-world LR images that are from your phone? If the real-world LR has the same artifact, it works. The color shift is normal since the training is a balance between content loss and gan loss in my understanding.

Thanks. 
Here is the real-world LR images that I took by phone camera.
https://drive.google.com/drive/folders/1Nv3TC3uN2-UuQlTVRtsGCp0beIdTtuzd?usp=sharing
I don't think there is artifact on that.
How can I solve the color shift problem?

**Comment by mrgreen3325** [comment]
> > > > > > > Have you retrained the DSN to reproduce the whole experiment? I think the first image is a good result for blind SR with unpaired images.
> > > > > > 
> > > > > > 
> > > > > > Hello. Have you try this? Is it successful to reproduce the whole experiment?
> > > > > 
> > > > > 
> > > > > No. In my understanding, you need to train the DSN carefully in order to generate LR that is degradation consistent with real-world LR. Only in this way, the SRN trained on the generated LR and HR pairs might have good generalization on real-world LR.
> > > > 
> > > > 
> > > > Thanks. Do you have any tricks to train the DSN well? I am training on it, however, the generated LR contains some artifact on it.
> > > 
> > > 
> > > What do you mean by "artifact" on LR?
> > > Thanks.
> > > Something like the line or dot on LR, and some color shift.
> > > ![01_116_3013](https://user-images.githubusercontent.com/25168884/174220717-1612cf09-82da-43f5-bc67-fee3ed454b48.png)
> > > My HR is face dataset, while LR is some images taken by my phone camera.
> 
> Can you show some real-world LR images that are from your phone? If the real-world LR has the same artifact, it works. The color shift is normal since the training is a balance between content loss and gan loss in my understanding.

Hi. Have you tried the DASR with nice output results?
So far, I tried to train several times and the results contain either artifacts or color shift.
